src/Tes.Runner/Logs/AppendBlobLogPublisher.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using Tes.Runner.Transfer;
+
+namespace Tes.Runner.Logs
+{
+    /// <summary>
+    /// Reads the log data from the stream and incrementally publishes it to Azure storage as append blobs.
+    /// </summary>
+    public class AppendBlobLogPublisher : StreamLogReader
+    {
+        private readonly string targetUrl;
+        private readonly BlobApiHttpUtils blobApiHttpUtils = new BlobApiHttpUtils();
+        private readonly string stdOutLogNamePrefix;
+        private readonly string stdErrLogNamePrefix;
+        private readonly ILogger logger = PipelineLoggerFactory.Create<AppendBlobLogPublisher>();
+
+        private int currentStdOutBLockCount;
+        private int currentStdErrBLockCount;
+
+        private string currentStdOutBlobName = string.Empty;
+        private string currentStdErrBlobName = string.Empty;
+
+
+        public AppendBlobLogPublisher(string targetUrl, string logNamePrefix)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(targetUrl);
+            ArgumentException.ThrowIfNullOrEmpty(logNamePrefix);
+
+            this.targetUrl = targetUrl;
+            var prefixTimeStamp = DateTime.UtcNow.ToString("yyyyMMddHHmmssfff");
+            stdOutLogNamePrefix = $"{logNamePrefix}_stdout_{prefixTimeStamp}";
+            stdErrLogNamePrefix = $"{logNamePrefix}_stderr_{prefixTimeStamp}";
+        }
+
+        private string GetBlobNameConsideringBlockCountCurrentState(int blockCount, string logName)
+        {
+            var blobNumber = blockCount / BlobSizeUtils.MaxBlobBlocksCount;
+
+            if (blobNumber == 0)
+            {
+                return $"{logName}.txt";
+            }
+
+            return $"{logName}_{blobNumber}.txt";
```

Comment:
we seem to be mixing underscore and hyphens with our file names... it seems like we should stick to hyphens since it's the canonical Linux naming convention?  Or in your mind are these different; the hypens are for multi-word names, and underscores are?

Reply:
The way I thought about this is: underscores separate terms/concepts that are not intended to be read together. Hyphens are a single-phrase name.